src/clients/SpokePoolClient.ts

```diff
@@ -334,6 +336,21 @@ export class SpokePoolClient extends BaseAbstractClient {
     return fills?.find((fill) => validateFillForDeposit(fill, deposit));
   }
 
+  private isEvmRepaymentValid(fill: Fill, matchedDeposit: Deposit): boolean {
+    // Slow fills don't result in repayments so they're always valid.
+    if (isSlowFill(fill)) {
+      return true;
+    }
+    // Lite chain deposits force repayment on origin chain.
+    const repaymentChainId = matchedDeposit.fromLiteChain ? fill.originChainId : fill.repaymentChainId;
+    // Return undefined if the requested repayment chain ID is not recognized by the hub pool.
+    const possibleRepaymentChainIds = this.configStoreClient?.getEnabledChains();
```

Comment:
We could alternatively assert(isDefined(this.configStoreClient)) but I'm worried this will make unit tests more cumbersome.